### PR TITLE
Log requests from site (close #1065)

### LINKED ISSUE #1065: Log requests from site
It's needed to test GA.
I think we may add `requests` property to session in Hammerhead. From test you would use it as `t.testRun.requests` - not documented feature.

/cc @inikulin

### PR DESCRIPTION
**Pull request by churkin**
/cc @inikulin @miherlosev

### CODE REVIEW

File: src/session/index.js [hunk 1]
```diff
@@ -135,12 +137,23 @@ export default class Session extends EventEmitter {
         };
     }
 
-    onPageRequest () {
-        if (this.requireStateSwitch) {
+    onRequest (ctx) {
+        var isPageRequest = ctx.isPage && !ctx.isIframe;
+
+        if (isPageRequest && this.requireStateSwitch) {
```
Comment (miherlosev) [inline src/session/index.js]: Add test for this enhancement

Comment (churkin) [inline src/session/index.js]: It is old code, i just move it form pipeline

**Comment by churkin** [comment]
@testcafe-build-bot /retest

**Comment by churkin** [comment]
Workaround for https://github.com/DevExpress/testcafe/issues/1306

**Pull request closed without merging**